sdk/typescript/src/types/sui-bcs.ts

```diff
@@ -431,6 +447,12 @@ const BCS_0_23_SPEC = {
       arguments: 'vector<CallArg>',
     },
   },
+  SharedObjectRef: {
```

Comment:
Your change is directionally correct - but now it's a bit complicated since we need to continue supporting TestNet(0.22.0).

In addition to adding your changes to `BCS_0_23_SPEC`, you also need to create a `BCS_0_24_SPEC` without  only the `SharedObjectRef` change(since the new `MoveCallTx` is supported in 0.24.0) and use this when rpc version == 0.24.0 on line 472